### PR TITLE
add support for warnings

### PR DESCRIPTION
**Pull request by lucabergamini**
replace prints with python warnings of different types

### CODE REVIEW

File: l5kit/l5kit/geometry/transform.py [hunk 1]
```diff
@@ -111,6 +112,8 @@ def transform_points(points: np.ndarray, transf_matrix: np.ndarray) -> np.ndarra
     Returns:
         np.ndarray: array of shape (N,2) for 2D input points, or (N,3) points for 3D input points
     """
+    if len(points.shape) == 1:
```
Comment (oliver-scheel) [inline l5kit/l5kit/geometry/transform.py:115]: Why was this changed?

Comment (lucabergamini) [reply l5kit/l5kit/geometry/transform.py:115]: because we're deprecating the function `transform_point` and this one will replace it. I should probably ensure the final shape is preserved though, what do you think?

Comment (lucabergamini) [reply l5kit/l5kit/geometry/transform.py]: I've removed this from the PR. I'll start a dedicated one targeting this one

**Review by oliver-scheel — APPROVED** [review]
LGTM
